pandas/core/arrays/datetimes.py

```diff
@@ -764,11 +762,10 @@ def _sub_datetimelike_scalar(self, other):
         if other is NaT:  # type: ignore[comparison-overlap]
             return self - NaT
 
-        if not self._has_same_tz(other):
-            # require tz compat
-            raise TypeError(
-                "Timestamp subtraction must have the same timezones or no timezones"
-            )
+        try:
+            self._assert_tzawareness_compat(other)
+        except TypeError as e:
```

Comment:
can you make this "err" instead of "e"; 1-letter variable names make grepping tough

Reply:
:+1: